src/nvim/fileio.c

```diff
@@ -3309,7 +3309,13 @@ static void vim_mktempdir(void)
     // "/tmp/" exists, now try to create "/tmp/nvim.<user>/".
     add_pathsep(tmp);
 
-    const char *appname = get_appname();
+    // Make a copy of the appname to avoid modifying it
+    char appname[40] = { 0 };
+    xstrlcpy(appname, get_appname(), sizeof(appname));
+    // Substitute slashes in appname with underscores
+    memchrsub(appname, '/', '_', sizeof(appname));
+    memchrsub(appname, '\\', '_', sizeof(appname));
```

Comment:
Should we use `%` since that is what is used for other things like undofiles ? Could also reduce the chances of accidental conflicts.

```suggestion
    memchrsub(appname, '/', '%', sizeof(appname));
    memchrsub(appname, '\\', '%', sizeof(appname));
```

Reply:
That makes sense, it's currently `_` because that's the same change that's made to [name, just above](https://github.com/neovim/neovim/blob/master/src/nvim/fileio.c#L3294-L3297).
If we were to change this here I would suggest changing both